docs/src/main/asciidoc/security-overview-concept.adoc

```diff
@@ -2,264 +2,33 @@
 = Quarkus Security overview
 include::_attributes.adoc[]
 :categories: security
-
+ 
 Quarkus Security is a framework that provides the architecture, multiple authentication and authorization mechanisms, and other tools for you to build secure and production-quality Java applications.
 
 == Getting started with Quarkus Security
 
-Before you start building security into your Quarkus applications, review the overview information to learn about the Quarkus Security architecture and the different authentication and authorization mechanisms that Quarkus supports.
+Before you start building security into your Quarkus applications, review the following resources to help you understand the security architecture and what's supported.
+
+* xref:security-architecture-concept.adoc[Quarkus Security architecture]
+* xref:security-authentication-mechanisms-concept.adoc[Supported authentication mechanisms in Quarkus]
 
 To get started with security in Quarkus, we recommend that you first combine the Quarkus built-in xref:security-basic-authentication-concept.adoc[Basic HTTP authentication] with the JPA identity provider to enable role-based access control (RBAC). 
 Complete the steps in the xref:security-basic-authentication-tutorial.adoc[Secure a Quarkus application with Basic authentication] tutorial.
 After you have successfully secured your Quarkus application with basic HTTP authentication, you can increase the security further by adding more advanced authentication mechanisms, for example, OpenID Connect (OIDC) authentication.
 
-== Security architecture
-
-The `HttpAuthenticationMechanism` interface is the main entry mechanism for securing HTTP applications in Quarkus.
-
-Quarkus Security uses `HttpAuthenticationMechanism` to extract the authentication credentials from the HTTP request and delegates them to `IdentityProvider` to convert the credentials to `SecurityIdentity`.
-For example, the credentials can come from the `Authorization` header, client HTTPS certificates, or cookies.
-
-`IdentityProvider` verifies the authentication credentials and maps them to `SecurityIdentity`, which has the username, roles, original authentication credentials, and other attributes.
-
-For every authenticated resource, you can inject a `SecurityIdentity` instance to get the authenticated identity information.
-
-In other contexts, it is possible to have other parallel representations of the same information or parts of it, for example, `SecurityContext`
-for JAX-RS or `JsonWebToken` for JSON Web Tokens (JWT).
-
-== Authentication mechanisms
```

Comment:
Note: All of this H2 section in security-overview-concept.adoc is now moved into a new concept doc file in this PR named security-authentication-mechanisms-concept.adoc.